File: Go/1 Arrays/Q6.go
// The program below counts and displays the frequency of each element in an array.
// The program accepts a specified number of integer inputs, stores them in an array, and then determines and prints how many times each element appears in the array.

package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)

	fmt.Println("Enter the number of elements to be stored in the array:")
	scanner.Scan()
	n, _ := strconv.Atoi(scanner.Text())

	arr := make([]int, n)
	fmt.Println("Enter the elements of the array:")
	for i := 0; i < n; i++ {
		fmt.Printf("element - %d: ", i+1)
		scanner.Scan()
		arr[i], _ = strconv.Atoi(scanner.Text())
	}

	frequency := make(map[int]int)
	for _, element := range arr {
		frequency[element]++
	}

	fmt.Println("\nThe frequency of all elements of the array:")
	for element, count := range frequency {
		fmt.Printf("%d occurs %d times\n", element, count)
	}
}

File: Rust/1 Arrays/Q6.rs
// The program below counts and displays the frequency of each element in an array. 
// The program accepts a specified number of integer inputs, stores them in an array, and then determines and prints how many times each element appears in the array.
// Author: Morteza Farrokhnejad
use std::collections::HashMap;
use std::io;

fn main() {
    let mut input = String::new();

    // Read the number of elements
    println!("Enter the number of elements to be stored in the array:");
    io::stdin().read_line(&mut input).expect("Failed to read input");
    let n: usize = input.trim().parse().expect("Please enter a valid number");
    input.clear();

    let mut arr = Vec::new();
    println!("Enter the elements of the array:");
    for i in 0..n {
        println!("element - {}:", i + 1);
        io::stdin().read_line(&mut input).expect("Failed to read input");
        let element: i32 = input.trim().parse().expect("Please enter a valid integer");
        arr.push(element);
        input.clear();
    }

    // Count the frequency of each element using a HashMap
    let mut frequency = HashMap::new();
    for &element in &arr {
        *frequency.entry(element).or_insert(0) += 1;
    }

    println!("\nThe frequency of all elements of the array:");
    for (element, count) in &frequency {
        println!("{} occurs {} times", element, count);
    }
}   